stumpy/motifs.py

```diff
@@ -465,11 +468,14 @@ def max_distance(D):
     else:
         candidate_idx = np.argmin(D)
 
-    while (
-        D[candidate_idx] <= atol + max_distance
-        and np.isfinite(D[candidate_idx])
-        and len(matches) < max_matches
-    ):
+    for i in range(len(D)):
+        if (
+            D[candidate_idx] > atol + max_distance
+            or D[candidate_idx] == np.inf
+            or len(matches) == max_matches
```

Comment:
For safety, I recommend `len(matches) >= max_matches`

Reply:
Sure. I will revise it.